### ISSUE TITLE
Appending charset to Content-Type

### ISSUE THREAD

**Issue by developerdong** [svenstaro/miniserve#263]
I have written Chinese as comments in a .py file. Miniserve can display this file online, but the Chinese characters are garbled like this:
```python
# 棣栧厛妫€鏌ュ弬鏁扮被鍨�
```

**Comment by KSXGitHub** [comment]
It works perfectly fine in my system (OS: Linux; Encoding: UTF-8)

![Screenshot](https://user-images.githubusercontent.com/11488886/77522431-d3847d00-6eb6-11ea-99d2-a25db6f03515.png)

What is your encoding?

**Comment by developerdong** [comment]
![image](https://user-images.githubusercontent.com/15864936/77625141-047bb500-6f7e-11ea-8820-734bdc1c34c0.png)
![image](https://user-images.githubusercontent.com/15864936/77625110-f3cb3f00-6f7d-11ea-88e2-ef6f29cd64f1.png)
The right characters are `这是一个中文字符串`，but the characters in broswer are `杩欐槸涓€涓腑鏂囧瓧绗︿覆`, it's wrong.

Miniserve is in a docker container which runing on Ubuntu 18.04.4, and I only set `LANG=zh_CN.UTF-8` environment in container.

**Comment by KSXGitHub** [comment]
This is the result I get without docker (`LANG=en_US.UTF-8`):

![Screenshot](https://user-images.githubusercontent.com/11488886/77626075-54a34900-6f77-11ea-9003-45e9af1194f6.png)

**Comment by KSXGitHub** [comment]
BTW, is there any reason your docker container's system language is not set to English? I am pretty sure a docker image is not your daily driver, so setting language is quite unnecessary.

**Comment by developerdong** [comment]
That's because I put my developement environment in docker, so that the migration of environment is easy. My native language is Chinese, I want to display and input Chinese in container. Back to this problem, garbled characters only appears in the browser on my Windows OS, it is normal in the terminal or downloaded to my desktop.

**Comment by KSXGitHub** [comment]
> garbled characters only appears in the browser on my Windows OS

Now I know why. Windows' default encoding (aka charset) isn't UTF-8. To fix this, miniserve may append "charset=utf-8" to `Content-Type` (e.g. `Content-Type: text/plain; charset=utf-8`).

I think you should change your title to "Appending charset to Content-Type" to make it easier for other contributors to know and fix your issue.

**Comment by KSXGitHub** [comment]
As a workaround, you may change your OS's or your browser's default encoding.

**Comment by aaronhatesregex** [comment]
To start with, the issue is caused by decoding UTF-8 encoded Chinese characters using ISO 8859-1 charset. I came across it today. (For TL;DR, see below; for suggestion, see last but one line.)

    Server
    miniserve version: 0.10.4
    OS: Debian GNU/Linux bullseye/sid x86_64
    Locale: en_US.UTF-8
    Target File Encoding: UTF-8

    Client 1
    Result: Wrong characters
    OS: Android
    Browser: Bromite (a fork of Chromium)
    Language preference (OS and browser): English (United States)
    
    Client 2
    Result: Wrong characters
    OS: Debian GNU/Linux bullseye/sid x86_64
    Browser: Firefox ESR
    Locale: en_US.UTF-8
    
    Client 3
    Result: Right characters
    OS: Debian and Termux on Android
    Browser: w3m
    Locale: en_US.UTF-8

TL;DR: The suggested workaround did not work.

I suppose it's because `Content-Type` HTTP header is usually used for `POST` and `PUT` requests but not `GET` according to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) and [Stack Overflow](https://stackoverflow.com/questions/5661596/do-i-need-a-content-type-header-for-http-get-requests).

I catch a `GET` request in Developer Tools in Firefox when **browsing a folder** and the response header includes `content-type: text/html; charset=utf-8`. But when I **browse a file**, the response header only includes `content-type: text/plain`.

Though `w3m http://127.0.0.1:8080/path/to/target.txt` works fine, I can reproduce the issue with `w3m -I iso-8859-1 http://127.0.0.1:8080/path/to/target.txt`. But I don't know how to set an ordinary GUI browser to include this head with every `GET` request (and it seems weird).

The man page of w3m reads:

    -I charset       document charset
    -O charset       display/output charset
        -I charset
            user defined character encoding of input data
        -O charset
            user defined character encoding of output data
        -T type
            explicit characterization of input data by MIME type

So my question is, can miniserve append `charset=utf-8` to `Content-Type` when making a request starting with `text/plain`?

I don't know much about Rust, but I notice in `listing.rs` there are `.content_type("text/html; charset=utf-8")` that can be related.

**Comment by svenstaro** [comment]
This should now be fixed. Give it a spin. :)

**Comment by svenstaro** [comment]
> Can miniserve append `charset=utf-8` to `Content-Type` when making a respond starting with `text/what_so_ever`? I came across this when serving a javascript file.

If you feel strongly about this, please open an issue. :)

**Comment by SuperKenVery** [comment]
Thanks for your suggestion, it's here: #1500
